### PR TITLE
ported docs for import_bgen and import_gen

### PR DESCRIPTION
**Pull request by jigold**
- Also added missing impex methods in sphinx docs and a reference to `default_reference` in docs.
- Moved API2 HailContext impex tests to methods tests.

### CODE REVIEW

File: python/hail/docs/api.rst [hunk 1]
```diff
@@ -47,4 +47,11 @@ on the Python programming interface.
...
     utils <utils/index>
     linalg <linalg/index>
-    stats <stats/index>
\ No newline at end of file
+    stats <stats/index>
+
+.. rubric:: Functions
+
+.. toctree::
+    :maxdepth: 1
```
Comment (patrick-schultz) [inline python/hail/docs/api.rst]: Those empty `toctree` blocks can all be removed. (Not that they do any harm.)

File: python/hail/methods/impex.py [hunk 2]
```diff
@@ -548,7 +552,95 @@ def grep(regex, path, max_count=100):
...
 def import_bgen(path, tolerance=0.2, sample_file=None, min_partitions=None, reference_genome=None,
                 contig_recoding=None):
-    rg = reference_genome if reference_genome else Env.hc().default_reference
+    """Import BGEN file(s) as matrix table.
+
+    Warning
+    -------
+
+    A BGEN file must have a ``.idx`` file which can be generated by
+    :func:`.index_bgen`
```
Comment (patrick-schultz) [inline python/hail/methods/impex.py]: Punctuation: 
```
A BGEN file must have a ``.idx`` file, which can be generated by
:func:`.index_bgen`.
```
Also, I don't understand what this is saying. Should it be ".idx file extension"?

File: python/hail/methods/impex.py [hunk 3]
```diff
@@ -548,7 +552,95 @@ def grep(regex, path, max_count=100):
...
+    only **unphased** and **diploid** genotype probabilities are allowed and the
+    genotype probability blocks must be either compressed with zlib or
+    uncompressed.
+
+    To load multiple files at the same time, use :ref:`Hadoop Glob Patterns
+    <sec-hadoop-glob>`.
+
+    **Column Fields**
+
+    - `s` (:class:`.TString`) -- Column key. This is the sample ID imported
```
Comment (patrick-schultz) [inline python/hail/methods/impex.py]: Most places in the docs I've seen, these bullet lists of fields are formatted with the field name bold: `*s*`. I don't think we've agreed on a style yet though.

File: python/hail/methods/impex.py [hunk 4]
```diff
@@ -548,7 +552,95 @@ def grep(regex, path, max_count=100):
...
+      from the first column of the sample file if given. Otherwise, the sample
+      id is taken from the sample identifying block in the first BGEN file if it
+      exists; else ids are assigned from `_0`, `_1`, to `_N`.
```
Comment (patrick-schultz) [inline python/hail/methods/impex.py]: "id" -> "ID" in both lines.

File: python/hail/methods/impex.py [hunk 5]
```diff
@@ -548,7 +552,95 @@ def grep(regex, path, max_count=100):
...
+        than the tolerance, set the entry to missing. Only applicable if the
+        BGEN files are v1.1.
+    sample_file : :obj:`str`, optional
+        Sample file to read the sample ids from. If specified, the number of
+        samples in the file must match the number in the BGEN file(s).
+    min_partitions : :obj:`int`, optional
+        Number of partitions.
+    reference_genome : :class:`.GenomeReference`, optional
+        Reference genome to use. Default is :func:`~hail2.default_reference`.
+        The row key will have type `TVariant(reference_genome)`.
```
Comment (patrick-schultz) [inline python/hail/methods/impex.py]: TVariant(reference_genome) should probably be in double backticks.

File: python/hail/methods/impex.py [hunk 6]
```diff
@@ -569,7 +661,85 @@ def import_bgen(path, tolerance=0.2, sample_file=None, min_partitions=None, refe
...
+    tolerance : :obj:`float`
+        If the sum of the genotype probabilities for a genotype differ from 1.0
+        by more than the tolerance, set the genotype to missing.
+    min_partitions : :obj:`int`, optional
+        Number of partitions.
+    chromosome : :obj:`str`, optional
+        Chromosome if not included in the GEN file
+    reference_genome : :class:`.GenomeReference`
+        Reference genome to use. Default is :func:`~hail2.default_reference`.
+        The row key will have type `TVariant(reference_genome)`.
```
Comment (patrick-schultz) [inline python/hail/methods/impex.py]: Likewise (double backticks).

File: python/hail/methods/impex.py [hunk 7]
```diff
@@ -366,14 +366,16 @@ def import_interval_list(path, reference_genome=None):
 
     reference_genome : :class:`.GenomeReference`
         Reference genome to use. Default is
-        :meth:`.HailContext.default_reference`.
+        :func:`~hail2.default_reference`.
```
Comment (tpoterba) [inline python/hail/methods/impex.py]: hail2 is gone

File: python/hail/methods/impex.py [hunk 8]
```diff
@@ -366,14 +366,16 @@ def import_interval_list(path, reference_genome=None):
...
 
     Returns
     -------
     :class:`.Table`
         Interval-keyed table.
     """
-    rg = reference_genome if reference_genome else Env.hc().default_reference
+
+    from hail2 import default_reference
```
Comment (tpoterba) [inline python/hail/methods/impex.py]: hail2 is gone

**Pull request merged**